### PR TITLE
Converting .then to async/await 

### PR DESCRIPTION
**Pull request by shikhar-scs**
As suggested by @marcoscaceres at https://github.com/w3c/respec/issues/1502#issuecomment-368250097 the `.then()` segments have been changed to `async` functions. @himanish-star had already filed a PR so his can be the one to be merged. However, I did this just for practice.

Please review.

### CODE REVIEW

**Comment by himanish-star** [comment]
@shikhar-scs, you're the best :heart:

File: src/core/issues-notes.js [hunk 1]
```diff
@@ -139,33 +139,44 @@ export function run(conf, doc, cb) {
...
+  
+  async function ghFetchAPI() {
+    try {
+      var JSONToBeSent = await ghFetch (conf.githubAPI)
+      var index = await getIndex (JSONToBeSent);
+      assignIndexToIssue (index);
+    } catch (err)  {
+      pub ("error", err.message);
+      handleIssues ($ins, ghIssues, issueBase);
+      cb ();
```
Comment (marcoscaceres) [inline src/core/issues-notes.js]: Note, "cb" is out of scope.

File: src/core/issues-notes.js [hunk 2]
```diff
@@ -139,33 +139,44 @@ export function run(conf, doc, cb) {
...
   }
+  
+  async function ghFetchAPI() {
+    try {
+      var JSONToBeSent = await ghFetch (conf.githubAPI)
+      var index = await getIndex (JSONToBeSent);
+      assignIndexToIssue (index);
+    } catch (err)  {
+      pub ("error", err.message);
+      handleIssues ($ins, ghIssues, issueBase);
```
Comment (marcoscaceres) [inline src/core/issues-notes.js]: note: $ins, ghIssues, issueBase are also all out of scope

File: src/core/issues-notes.js [hunk 3]
```diff
@@ -139,33 +139,44 @@ export function run(conf, doc, cb) {
...
+      cb ();
+    }
+  }
+  
+  function getIndex(json) {
+    issueBase = issueBase || json.html_url + "/issues/";
```
Comment (marcoscaceres) [inline src/core/issues-notes.js]: reassignment of the outer scope `issueBase` could be confusing - why not assign a new var here instead?

File: src/core/issues-notes.js [hunk 4]
```diff
@@ -139,33 +139,44 @@ export function run(conf, doc, cb) {
...
+      // See: https://developer.github.com/v3/media/
+      Accept: "application/vnd.github.v3.html+json",
+    });
+  }
+  
+  function assignIndexToIssue(issues) {
+    issues.forEach (function (issue) {
+      ghIssues[issue.number] = issue;
+    });
+    handleIssues ($ins, ghIssues, issueBase);
```
Comment (marcoscaceres) [inline src/core/issues-notes.js]: This function seems to do more work than it needs to. I'd suggest taking `handleIssues()` and `cb()` out of this function.

File: src/core/issues-notes.js [hunk 5]
```diff
@@ -139,33 +139,44 @@ export function run(conf, doc, cb) {
...
+    issueBase = issueBase || json.html_url + "/issues/";
+    return fetchIndex (json.issues_url, {
+      // Get back HTML content instead of markdown
+      // See: https://developer.github.com/v3/media/
+      Accept: "application/vnd.github.v3.html+json",
+    });
+  }
+  
+  function assignIndexToIssue(issues) {
+    issues.forEach (function (issue) {
```
Comment (marcoscaceres) [inline src/core/issues-notes.js]: Nit: this should probably be a `.reduce()`

File: src/core/issues-notes.js [hunk 6]
```diff
@@ -139,33 +139,44 @@ export function run(conf, doc, cb) {
...
+  function getIndex(json) {
+    issueBase = issueBase || json.html_url + "/issues/";
+    return fetchIndex (json.issues_url, {
+      // Get back HTML content instead of markdown
+      // See: https://developer.github.com/v3/media/
+      Accept: "application/vnd.github.v3.html+json",
+    });
+  }
+  
+  function assignIndexToIssue(issues) {
```
Comment (marcoscaceres) [inline src/core/issues-notes.js]: assignIndexToIssue might not need to exist:

```JS
var issues = await getIndex (JSONToBeSent);
issues.reduce((collector, issue ) => {
  ghIssues[issue.number] = issue;
  return collector;
} , ghIssues )
```

**Comment by marcoscaceres** [comment]
Not bad... try not to nest too much functionality into functions. Also, try no to rely on outer scope variables so much. That can lead to really hard bugs, specially here where we are dealing with async code.

**Comment by marcoscaceres** [comment]
Closing - as this was just a coding exercise. Thanks for this input @shikhar-scs!

**Comment by shikhar-scs** [comment]
Thank you @marcoscaceres for all the informative inputs. I'm over with my exams and thus would be starting with all the work now.

**Pull request closed without merging**